OpenSim/Simulation/SimbodyEngine/Joint.cpp

```diff
@@ -56,34 +57,70 @@ Joint::Joint() : Super()
     constructInfrastructure();
 }
 
-/**
- * API constructor.
- */
-Joint::Joint(const std::string &name, 
-    const PhysicalFrame& parent, 
-    const SimTK::Vec3& locationInParent, const SimTK::Vec3& orientationInParent,
-    const PhysicalFrame& child, 
-    const SimTK::Vec3& locationInChild, const SimTK::Vec3& orientationInChild, 
-    bool reverse) : Super()
+/* API constructor. */
+Joint::Joint(const std::string &name, const std::string& parentName, 
+                                      const std::string& childName,
+                                      bool reverse) : Joint()
 {
-    setNull();
-    constructInfrastructure();
-
-    set_location_in_parent(locationInParent);
-    set_orientation_in_parent(orientationInParent);
-    set_location_in_child(locationInChild);
-    set_orientation_in_child(orientationInChild);
+    setName(name);
     set_reverse(reverse);
 
-    updConnector<PhysicalFrame>("parent_body").set_connectee_name(parent.getName());
-    updConnector<PhysicalFrame>("child_body").set_connectee_name(child.getName());
+    updConnector<PhysicalFrame>("parent_frame").set_connectee_name(parentName);
+    updConnector<PhysicalFrame>("child_frame").set_connectee_name(childName);
+}
+
+/* Deprecated Constructor*/
```

Comment:
This seems wimpy. If it isn't deprecated I'd take this out and just call this a convenience constructor for the common case where the joint will own the two Frames it uses to attach to its bodies.